incubator/hnc/internal/validators/anchor_test.go

```diff
@@ -54,3 +49,53 @@ func TestSubnamespaces(t *testing.T) {
 		})
 	}
 }
+
+func TestAllowCascadingDeleteSubnamespaces(t *testing.T) {
+	// Create a chain of namespaces from "a" to "e", with "a" as the root. Among them,
+	// "b", "d" and "e" are subnamespaces. This is set up in a long chain to test that
+	// subnamespaces will look all the way up to get the 'allowCascadingDelete` value
+	// and won't stop looking when the first full namespace ancestor is met.
+	f := foresttest.Create("-AbCD")
+	h := &Anchor{Forest: f}
+
+	tests := []struct {
+		name string
+		op   v1beta1.Operation
+		acd  []string
+		pnm  string
+		cnm  string
+		fail bool
+	}{
+		{name: "ok - delete subnamespace with allowCascadingDelete set in the parent namespace", op: v1beta1.Delete, acd: []string{"c"}, pnm: "c", cnm: "d"},
```

Comment:
Nit - I'd take out the `[not] ok - delete subnamespace with` prefix. Whether it's ok or not is encoded in the `fail` and the expected value will be printed out if there's a problem, and the test is called `TestAllowCascadingDeleteSubnamespaces` so we're just repeating that.

I'd suggest calling these something like:
`set in parent`
`set in non-leaf`
`set in ancestor that is not the first full namepsace`
`unset in leaf`
`unset in non-leaf`